server/src/main/java/de/uftos/repositories/solver/timefold/constraints/ConstraintDefinitionFactory.java

```diff
@@ -0,0 +1,1073 @@
+      }
+
+      return ((Number) first).getValue() > ((Number) second).getValue();
+    };
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Boolean> convertIn(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.IN) {
+      throw new IllegalStateException();
+    }
+    OperatorDto in = (OperatorDto) ast;
+    if (in.parameters().size() != 2) {
+      throw new IllegalStateException();
+    }
+    Function<List<ResourceTimefoldInstance>, ResourceTimefoldInstance> element =
+        convertElement(in.parameters().getFirst(), params);
+    Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> set =
+        convertSet(in.parameters().getLast(), params);
+    return (parameters) -> set.apply(parameters).contains(element.apply(parameters));
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> convertOf(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.OF) {
+      throw new IllegalStateException();
+    }
+
+    OperatorDto of = (OperatorDto) ast;
+
+    if (of.parameters().size() != 2) {
+      throw new IllegalStateException();
+    }
+
+    AbstractSyntaxTreeDto set = of.parameters().getLast();
+
+    return convertSet(set, params);
+  }
+
+  private static Function<
+      List<ResourceTimefoldInstance>,
+      Function<Set<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+      > convertFilter(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.FILTER) {
+      throw new IllegalStateException();
+    }
+
+    OperatorDto filter = (OperatorDto) ast;
+
+    List<Function<List<ResourceTimefoldInstance>, Boolean>> functions = new ArrayList<>();
+
+    for (AbstractSyntaxTreeDto param : filter.parameters()) {
+      if (param.getToken() == UcdlToken.NUMBER_SET || param.getToken() == UcdlToken.RESOURCE_SET) {
+        Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> set =
+            convertSet(param, params);
+        Function<List<ResourceTimefoldInstance>, ResourceTimefoldInstance> element =
+            convertValueReference(new ValueDto<>(UcdlToken.VALUE_REFERENCE, "this"), params);
+        functions.add((parameters -> set.apply(parameters).contains(element.apply(parameters))));
+      } else {
+        functions.add(convertBool(param, params));
+      }
+    }
+
+    return (parameters -> set -> {
+      ResourceTimefoldInstance thisPointer = parameters.getFirst();
+      Set<ResourceTimefoldInstance> filteredSet = new HashSet<>();
+      outerLoop:
+      for (ResourceTimefoldInstance resource : set) {
+        parameters.set(0, resource); //replacing "this"-pointer
+        for (Function<List<ResourceTimefoldInstance>, Boolean> function : functions) {
+          if (!function.apply(parameters)) {
+            continue outerLoop;
```

Comment:
Uuuuhm. Can't you just`break` out of the inner loop? Goto is honestly extremely ugly and prone to confusion

Reply:
Only breaking the inner loop will still result in `filteredSet.add(resource);` being executed which it isn`t supposed to.

Reply:
I see. Then I may add a variable for keeping state or something. Idk.... just *anything* else to not have jump targets lol

Reply:
I guess thats doable.